library/modules/bigip_snat_translation.py

```diff
@@ -0,0 +1,694 @@
+        'disabled',
+        'enabled',
+        'ip_idle_timeout',
+        'state',
+        'tcp_idle_timeout',
+        'traffic_group',
+        'udp_idle_timeout'
+    ]
+
+    updatables = [
+        'arp',
+        'connection_limit',
+        'description',
+        'disabled',
+        'enabled',
+        'ip_idle_timeout',
+        'traffic_group',
+        'tcp_idle_timeout',
+        'udp_idle_timeout'
+    ]
+
+    @property
+    def traffic_group(self):
+        if self._values['traffic_group'] is None:
+            return None
+        else:
+            result = fq_name(self.partition, self._values['traffic_group'])
+
+
+class ApiParameters(Parameters):
+    @property
+    def connection_limit(self):
+        if self._values['connection_limit'] is None:
+            return None
+        return int(self._values['connection_limit'])
+
+    @property
+    def enabled(self):
+        if 'enabled' in self._values:
+            return True
+        return False
+
+    @property
+    def disabled(self):
+        if 'disabled' in self._values:
+            return True
+        return False
+
+
+class ModuleParameters(Parameters):
+    def _clear_member_prefix(self, address):
+        result = os.path.basename(address)
+        return result
+
+    def _format_ip_address(self, address):
+        if is_valid_ip(address):
+            address = '/{0}/{1}'.format(self.partition, address)
+            return address
+        else:
+            raise F5ModuleError(
+                'The provided member address is not a valid IP address'
+            )
+
+    @property
+    def arp(self):
+        result = flatten_boolean(self._values['arp'])
+        if result is None:
+            return None
+        if result == 'yes':
+            return 'enabled'
+        return 'disabled'
+
+    @property
+    def connection_limit(self):
+        if self._values['connection_limit'] is None:
+            return None
+        return int(self._values['connection_limit'])
+
+    @property
+    def description(self):
+        if self._values['description'] in [None, 'none']:
+            return None
+        return self._values['description']
+
+    @property
+    def enabled(self):
+        if self._values['state'] in ['enabled', 'present']:
+            return True
+
+    @property
+    def state(self):
+        if self.enabled is True and self._values['state'] != 'present':
+            return 'enabled'
+        elif self.disabled is True:
+            return 'disabled'
+        else:
+            return self._values['state']
+
+
+class Changes(Parameters):
+    def to_return(self):
+        result = {}
+        try:
+            for returnable in self.returnables:
+                result[returnable] = getattr(self, returnable)
+            result = self._filter_params(result)
+        except Exception:
+            pass
+        return result
+
+
+class UsableChanges(Changes):
+    pass
+
+
+class ReportableChanges(Changes):
+    pass
```

Comment:
are the returned values the same as user input them? you would want to do something like this if they are not, obviously not the same code but this is to get you the idea:
https://github.com/F5Networks/f5-ansible/blob/devel/library/modules/bigip_message_routing_router.py#L427